pkgs/applications/audio/musescore/default.nix

```
@@ -83,6 +83,8 @@ in stdenv'.mkDerivation (finalAttrs: {
     # Don't bundle qt qml files, relevant really only for darwin, but we set
     # this for all platforms anyway.
     "-DMUE_COMPILE_INSTALL_QTQML_FILES=OFF"
+    # Don't build unit tests unless we are going to run them.
+    "-DMUSE_ENABLE_UNIT_TESTS=${if finalAttrs.doCheck then "ON" else "OFF"}"
```

Hmm that's a dilemma I faced a few times in the past. Usually the rule of thumb I go with here is: "Testing that unit tests build, is useful even when running them is disabled, because that's a test within itself". That argument is relevant mostly when the package might be cross compiled.

However in this case, I'm not sure - we never run these tests, even on native builds, so IDK... In anycase:

```suggestion
    # Don't build unit tests unless we are going to run them.
    (lib.cmakeBool "MUSE_ENABLE_UNIT_TESTS" finalAttrs.doCheck)
```

---

I am applying the intention of cmake setup-hook:
https://github.com/NixOS/nixpkgs/blob/24.05/pkgs/by-name/cm/cmake/setup-hook.sh#L105-L108

---

> I am applying the intention of cmake setup-hook: [`24.05`/pkgs/by-name/cm/cmake/setup-hook.sh#L105-L108](https://github.com/NixOS/nixpkgs/blob/24.05/pkgs/by-name/cm/cmake/setup-hook.sh?rgh-link-date=2024-09-16T14%3A07%3A50Z#L105-L108)

OK. if this was a more low-tree package that might have been cross compiled I would have insisted to leave it as is, but no one will cross compile it probably, and even if they will it shouldn't matter that much.